autosklearn/ensembles/multiobjective_dummy_ensemble.py

```diff
@@ -0,0 +1,458 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union
+
+import warnings
+
+import numpy as np
+
+from autosklearn.automl_common.common.utils.backend import Backend
+from autosklearn.constants import TASK_TYPES
+from autosklearn.data.validation import SUPPORTED_FEAT_TYPES
+from autosklearn.ensemble_building.run import Run
+from autosklearn.ensembles.abstract_ensemble import (
+    AbstractEnsemble,
+    AbstractMultiObjectiveEnsemble,
+)
+from autosklearn.metrics import Scorer, calculate_losses
+from autosklearn.pipeline.base import BasePipeline
+
+
+class SingleModelEnsemble(AbstractEnsemble):
+    """Ensemble consisting of a single model.
+
+    This class is used my the :cls:`MultiObjectiveDummyEnsemble` to represent ensembles
+    consisting of a single model, and this class should not be used on its own.
+
+    Parameters
+    ----------
+    task_type: int
+        An identifier indicating which task is being performed.
+
+    metrics: Sequence[Scorer] | Scorer
+        The metrics used to evaluate the models.
+
+    random_state: Optional[int | RandomState] = None
+        Not used.
+
+    backend : Backend
+        Gives access to the backend of Auto-sklearn. Not used.
+
+    model_index : int
+        Index of the model that constitutes the ensemble. This index will
+        be used to select the correct predictions that will be passed during
+        ``fit`` and ``predict``.
+    """
+
+    def __init__(
+        self,
+        task_type: int,
+        metrics: Sequence[Scorer] | Scorer,
+        random_state: int | np.random.RandomState | None,
+        backend: Backend,
+        model_index: int,
+    ):
+        # Add some default values -- at least 1 model in ensemble is assumed
+        self.indices_ = [model_index]
+        self.weights_ = [1.0]
+        self.task_type = task_type
+        self.metrics = metrics if isinstance(metrics, Sequence) else [metrics]
+
+    def fit(
+        self,
+        base_models_predictions: np.ndarray | List[np.ndarray],
+        X_data: SUPPORTED_FEAT_TYPES,
```

Comment:
`X_data` can be None I think? Again, not sure if it should have that as a default

Reply:
Yes, added this. WRT whether it should have a default, I would say no, using the same argument as above.